lib/diaspora_federation/entity.rb

```diff
@@ -319,8 +319,8 @@ def simple_node(doc, name, value)
       when :integer
         text.to_i if text =~ /^\d+$/
       when :boolean
-        return true if text =~ /(true|t|yes|y|1)$/i
-        false if text =~ /(false|f|no|n|0)$/i
+        return true if text =~ /^(true|t|yes|y|1)$/i
+        false if text =~ /^(false|f|no|n|0)$/i
```

Comment:
Careful here, in Ruby `^`and `$`are beginning/end of **line** respectively, regardless of `/m`. If Beginning/end of **string** is intended here, use `\A`and `\z`respectively.